scripts/postinstall.js

```diff
@@ -80,6 +119,8 @@ async function main() {
   }
 
   const pkg = await readPackageJson();
+  const hash = createHash("sha256");
+  const expectedChecksum = await parseCheckSumFile(pkg);
```

Comment:
If checksum for a binary is not found, `expectedChecksum` will be undefined and the subsequent comparison with calculated checksum will also fail. I think it's a bit harsh to fail the installation completely if no checksum is found.

How about we change `parseCheckSumFile` to return a dictionary of filename => checksum, and only compares the checksum if the filename exists in the dictionary?

Reply:
Not finding the checksum for a file would be very unlike as it's all generated by the GitHub releases artifacts and is available in the checksum text file.

So, don't you think that the checksum would always be found?
Also I generated a dictionary earlier but thought keeping that much data is useles as we will only be comparing checksum for one file.
And about failing the installation I would say isn't that the whole point if veryfying the checksums? 
To fail the installation if verification fails

Reply:
I don't mind making the changes you said,
It's just I am thinking that it would be very abnormal situation when we can't find the checksum in the checksum file from GitHub artifacts...
So final decision is on you
I'll make the changes right away if you say so

Reply:
There are two cases here that I think would be nice to handle separately

1. If checksum doesn't exist => log a warning and continue installation
2. if checksum exists but mismatch => fail the installation

The current implementation handles the 2nd case well. The 1st case could happen when checksum fails to download due to network issue, or may be goreleaser fails to upload the checksum for a specific platform, or some other unknown scenario.

I agree those scenarios don't happen often. What I'm hoping to avoid though is a hard dependency on `checksum.txt` before downloading any binary via npm.

Reply:
ahhh huh i see your point now.
Fair enough
give me 2 mins i'll update the PR

Reply:
The changes have been done and now if the file is not fetched then it looks like this :- 

![Screenshot from 2023-08-22 15-49-55](https://github.com/supabase/cli/assets/73625149/c2714e41-4229-4202-85ea-0519e963ad9e)

and when all works out this :-

![image](https://github.com/supabase/cli/assets/73625149/8a0b0510-a8f4-4d85-91db-a9e24c428713)

Reply:
and yes for testing purposes and simulating a scenario where file is not fetched i intentionally wrote the wrong url for checksums in first image. 😃

Reply:
Nice, this is greatly appreciated!

Reply:
Thank you😁
So it's mergeable now ig